### DISCUSSION TITLE
Connect provider to hardhat built-in network

### DISCUSSION THREAD

**Discussion by ghost** [NomicFoundation/hardhat#3076] [Troubleshooting] [answered]
>There are two kinds of networks in Hardhat: [JSON-RPC](https://eth.wiki/json-rpc/API) based networks, and the built-in Hardhat Network.

I know I can use localhost with `hardhat node`, but is it possible to connect ethers provider to built-in network like this?:
`const provider = new ethers.providers.JsonRpcProvider(builtin_network_url);`
I want to use pure *ethers.js* library to deploy contract but with built-in network, so I don't need to restart node every time.
With *hardhat-ethers* it's `ethers.provider` object, but without this plugin?

If I do this:
``` solidity
const provider = new ethers.providers.JsonRpcProvider();
const deployer = new ethers.Wallet(
    process.env.ADMIN_PRIVATE_KEY!,
    provider
);
console.log(provider.connection);
console.log(await deployer.getBalance());
```
I get:
![image](https://user-images.githubusercontent.com/97038258/186617401-a3126707-31b0-4c80-9e24-48206ae6778c.png)

**Comment by fvictorio** [comment, marked as answer]
This is not possible because the in-process network doesn't open an HTTP connection. So your options are:

1. Use `ethers.provider` from the `hardhat-ethers` plugin
2. Wrap the `network.provider` with [`ethers.Web3Provider`](https://docs.ethers.io/v5/single-page/#/v5/api/providers/other/-%23-Web3Provider). `network.provider` is an EIP-1193 provider so you can pass it to that ethers's provider.

**Reply by ghost** [reply]
`const prov = new ethers.providers.Web3Provider(network.provider)` ?
*Argument of type 'EthereumProvider' is not assignable to parameter of type 'ExternalProvider | JsonRpcFetchFunc'.
  Type 'EthereumProvider' is not assignable to type 'ExternalProvider'.
    Types of property 'sendAsync' are incompatible.*

**Reply by fvictorio** [reply]
Oh, I don't know why it doesn't compile. But I'm pretty sure it should work fine. I'd recommend you do `(network.provider as any)`, check if it works, and then decide if losing type-safety there is worth it.

**Reply by ghost** [reply]
Seems it works. Maybe Ill use it to deploy my contract with ethers as for practice. For main projects I gonna definitely use *hardhat-ethers*. Thanks!
